### PR TITLE
coprocessor: Add IsNull, IsTrue, IsFalse RPN functions

### PR DESCRIPTION
**Pull request by sticnarf**
## What have you changed? (mandatory)

This PR adds IsNull for Int, Real, Decimal, String, Time, Duration and Json. It also adds IsTrue and IsFalse for Int, Decimal and Real.

I am not sure if we may have a column containing `Json::None` and should we treat it as null. In this PR, a `Json::None` is not null.

## What are the type of the changes? (mandatory)

Improvement (Support these functions in batch)

## How has this PR been tested? (mandatory)

Unit tests.

## Does this PR affect documentation (docs) or release note? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

**Comment by breeswish** [comment]
Great job!

BTW your implementation is correct and doesn't need to concern about the JSON's null:

```
mysql> select null is null;
+--------------+
| null is null |
+--------------+
|            1 |
+--------------+
1 row in set (0.04 sec)

mysql> select JSON_EXTRACT('{"hello": null}', '$.hello') is null;
+----------------------------------------------------+
| JSON_EXTRACT('{"hello": null}', '$.hello') is null |
+----------------------------------------------------+
|                                                  0 |
+----------------------------------------------------+
1 row in set (0.02 sec)

mysql> select JSON_EXTRACT('{"hello": null}', '$.hello');
+--------------------------------------------+
| JSON_EXTRACT('{"hello": null}', '$.hello') |
+--------------------------------------------+
| null                                       |
+--------------------------------------------+
1 row in set (0.01 sec)
```

File: src/coprocessor/dag/rpn_expr/mod.rs [hunk 1]
```diff
@@ -98,6 +98,19 @@ fn map_pb_sig_to_rpn_func(value: ScalarFuncSig, children: &[Expr]) -> Result<Box
...
+        ScalarFuncSig::StringIsNull => Box::new(RpnFnIsNull::<Bytes>::new()),
+        ScalarFuncSig::TimeIsNull => Box::new(RpnFnIsNull::<DateTime>::new()),
+        ScalarFuncSig::DurationIsNull => Box::new(RpnFnIsNull::<Duration>::new()),
+        ScalarFuncSig::IntIsTrue => Box::new(RpnFnIntIsTrue),
+        ScalarFuncSig::RealIsTrue => Box::new(RpnFnRealIsTrue),
+        ScalarFuncSig::DecimalIsTrue => Box::new(RpnFnDecimalIsTrue),
+        ScalarFuncSig::IntIsFalse => Box::new(RpnFnIntIsFalse),
+        ScalarFuncSig::RealIsFalse => Box::new(RpnFnRealIsFalse),
+        ScalarFuncSig::DecimalIsFalse => Box::new(RpnFnDecimalIsFalse),
+        ScalarFuncSig::JsonIsNull => Box::new(RpnFnIsNull::<Json>::new()),
```
Comment (breeswish) [inline src/coprocessor/dag/rpn_expr/mod.rs]: You may put JsonIsNull at Line 107.

Comment (sticnarf) [reply src/coprocessor/dag/rpn_expr/mod.rs]: Fixed.

File: src/coprocessor/dag/rpn_expr/impl_op.rs [hunk 2]
```diff
@@ -53,10 +54,128 @@ impl RpnFnLogicalOr {
...
+pub struct RpnFnIntIsTrue;
+
+impl RpnFnIntIsTrue {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg: &Option<Int>,
+    ) -> Result<Option<i64>> {
+        Ok(*arg)
```
Comment (breeswish) [inline src/coprocessor/dag/rpn_expr/impl_op.rs]: it should be `v != 0i64`?

Comment (sticnarf) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: The behavior of `is true` for Int is the same as evaluating the int as mysql bool, so I just return the arg.
Maybe I should add a comment or use the evident way?

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: @zz-jason PTAL. Please help verify in what scenario this function will be used. I tried some expressions in TiDB and according to EXPLAIN this function does not occur 🤔

Comment (lonng) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: See: https://github.com/pingcap/tidb/blob/d9e02b76ddb7a99d6740728edb58612d5b617b0f/expression/builtin_op.go#L475

Comment (lonng) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: I think it's better to keep consistent with TiDB.

**Comment by sticnarf** [comment]
I'm sorry but I find the implementation incorrect. IsTrue and IsFalse should not return None if the input is None.

**Comment by sticnarf** [comment]
@breeswish @lonng Please take a look again

**Pull request merged**